trinity/sync/common/chain.py

```diff
@@ -306,6 +349,7 @@ def get_target_header_hash(self) -> Hash32:
 
             yield headers
             last_received_header = headers[-1]
+            self.sync_progress.current_block = last_received_header.block_number
```

Comment:
See my comment below about named tuples.  Instead of updating the existing object (and thus mutating it), I'd prefer we created an all new `SyncProgress` item.

Reply:
Hm, any reason we are afraid of mutating objects in this project? Introducing setter is ok, but recreating whole object.. why?

Reply:
We have a reasonably strong convention across all of our codebases to avoid mutability unless faced with a specific reason why it is necessary.  This gets us two objective benefits.

- Reduced cognitive overhead.  When reading non mutative code, you don't have to track state which makes reading the code significantly easier.
- Avoiding multiple types of bugs entirely.

So, while in many cases, a mutation based approach may be fine, we still opt to find a non-mutative way.

Reply:
Well, in both cases we change memory behind self.sync_progress variable, complexity of tracking this change in code is same for both approaches. The only difference is changing only one values vs all three values at once, two of which are always redundant. I hardly can see any benefits but wasting of CPU cycles here. However, if it's a "holy rule" of a project - I understand and can follow it.

Reply:
I understand that it may seem un-necessary, and it might be, but I'd like it updated to remove the mutation.

Reply:
>Well, in both cases we change memory behind self.sync_progress variable, complexity of tracking this change in code is same for both approaches

Not entirely. Some code path could hold on to an instance of `SyncProgress` and do something with it based on the state of its properties. And with the solution that uses mutation, these properties could just change which basically means, you have to keep in mind that they could technically just change from one moment to the next.

In the non-mutation version, a code path that holds on to a`SyncProgress` object never has to worry about its state changing. It will always stay the way it is.